storage-operator/pkg/apis/storage/v1alpha1/volume_types_test.go

```diff
@@ -0,0 +1,82 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var TEST_VOLUMES = map[string]*Volume{
+	"SparseLoopDevice": &Volume{
+		ObjectMeta: metav1.ObjectMeta{UID: "F00"},
+		Spec: VolumeSpec{
+			VolumeSource: VolumeSource{
+				SparseLoopDevice: &SparseLoopDeviceVolumeSource{
+					Size: resource.MustParse("1Gi"),
+				},
+			},
+		},
+	},
+	"RawBlockDevice": &Volume{
+		ObjectMeta: metav1.ObjectMeta{UID: "BAR"},
+		Spec: VolumeSpec{
+			VolumeSource: VolumeSource{
+				RawBlockDevice: &RawBlockDeviceVolumeSource{
+					DevicePath: "/dev/sda",
+				},
+			},
+		},
+	},
+}
+
+func TestGetSize(t *testing.T) {
+	tests := map[string]struct {
+		disk_size resource.Quantity
+		expected  resource.Quantity
+	}{
+		"SparseLoopDevice": {
+			disk_size: resource.MustParse("100Gi"),
+			expected:  resource.MustParse("1Gi"),
+		},
+		"RawBlockDevice": {
+			disk_size: resource.MustParse("100Gi"),
+			expected:  resource.MustParse("100Gi"),
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			volume := TEST_VOLUMES[name]
+
+			size := volume.GetSize(tc.disk_size)
+
+			assert.Equal(t, tc.expected, size)
+		})
+	}
+}
+
+func TestGetPath(t *testing.T) {
+	tests := map[string]struct {
+		path     string
+		expected string
+	}{
+		"SparseLoopDevice": {
+			path:     "/dev/sdb",
```

Comment:
This `path` field seems unused in the test.

Reply:
Good catch. I'll remove it.